vault-plugins/flant_iam/backend/path_identitysharing.go

```diff
@@ -0,0 +1,225 @@
+			Fields: map[string]*framework.FieldSchema{
+				"uuid": {
+					Type:        framework.TypeNameString,
+					Description: "ID of an identity sharing",
+					Required:    true,
+				},
+				"tenant_uuid": {
+					Type:        framework.TypeNameString,
+					Description: "ID of a tenant",
+					Required:    true,
+				},
+			},
+			ExistenceCheck: b.handleExistence(),
+			Operations: map[logical.Operation]framework.OperationHandler{
+				// do we need it? Maybe we can change groups...
+				// logical.UpdateOperation: &framework.PathOperation{
+				// 	Callback: b.handleUpdate(),
+				// 	Summary:  "Update the user by ID",
+				// },
+				logical.ReadOperation: &framework.PathOperation{
+					Callback: b.handleRead,
+					Summary:  "Retrieve the identity sharing by ID",
+				},
+				logical.DeleteOperation: &framework.PathOperation{
+					Callback: b.handleDelete,
+					Summary:  "Deletes the identity sharing by ID",
+				},
+			},
+		},
+		// Listing
+		{
+			Pattern: "tenant/" + uuid.Pattern("tenant_uuid") + "/identity_sharing/?",
+			Fields: map[string]*framework.FieldSchema{
+				"tenant_uuid": {
+					Type:        framework.TypeNameString,
+					Description: "ID of a tenant",
+					Required:    true,
+				},
+			},
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ListOperation: &framework.PathOperation{
+					Callback: b.handleList,
+					Summary:  "Lists all tenant identity sharing IDs.",
+				},
+			},
+		},
+	}
+}
+
+func (b *identitySharingBackend) handleCreate(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	sourceTenant := data.Get("tenant_uuid").(string)
+	destTenant := data.Get("destination_tenant_uuid").(string)
+	groups := data.Get("groups").([]string)
+
+	tx := b.storage.Txn(true)
+	defer tx.Abort()
+	repo := model.NewIdentitySharingRepository(tx)
+
+	is := &model.IdentitySharing{
+		UUID:                  uuid.New(),
+		SourceTenantUUID:      sourceTenant,
+		DestinationTenantUUID: destTenant,
+		Groups:                groups,
+	}
+
+	if err := repo.Create(is); err != nil {
+		msg := "cannot create identity sharing"
+		b.Logger().Debug(msg, "err", err.Error())
+		return logical.ErrorResponse(msg), nil
+	}
+
+	if err := commit(tx, b.Logger()); err != nil {
+		return nil, err
+	}
+
+	resp := &logical.Response{Data: map[string]interface{}{"identity_sharing": is}}
+	return logical.RespondWithStatusCode(resp, req, http.StatusCreated)
+}
+
+func (b *identitySharingBackend) handleList(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	sourceTenant := data.Get("tenant_uuid").(string)
+
+	tx := b.storage.Txn(false)
+	defer tx.Abort()
+
+	repo := model.NewIdentitySharingRepository(tx)
+
+	list, err := repo.List(sourceTenant)
+	if err != nil {
+		return nil, err
+	}
+
+	_ = commit(tx, b.Logger())
+
+	resp := &logical.Response{
+		Data: map[string]interface{}{
+			"uuids": list,
```

Comment:
We should have more verbose objects here

Reply:
done